### PR TITLE
feat: github actions to publish ecr and docker

### PR DESCRIPTION
**Pull request by krol3**
github actions to publish ecr and docker

Close https://github.com/aquasecurity/trivy-db/issues/31

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 FROM golang:1.15-alpine as builder
 
-ARG DB_TYPE=trivy
+ARG DB_TYPE=trivy-light
```
Comment (knqyf263) [inline Dockerfile:3]: Why did you change it?

**Review by knqyf263 — COMMENTED** [review]
Did you test it in your fork or something else?

**Comment by krol3** [comment]
> Did you test it in your fork or something else?

Yes, you can validate it [here](https://github.com/krol3/trivy-db/actions)

**Comment by knqyf263** [comment]
>Yes, you can validate it here

Nice!

**Review by knqyf263 — COMMENTED** [review]
Seems `coverage.txt` is included by mistake.

File: .github/workflows/cron.yml [hunk 2]
```diff
@@ -1,66 +1,64 @@
+---
```
[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/cron.yml:1]: This is yaml lint, it's the default rule for [document start](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.document_start)

File: .github/workflows/publish-ecr.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install AWS Cli 2.0
+        run: |
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
```
[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/publish-ecr.yml:22]: Nice! I will test it, and give you an answer

[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/publish-ecr.yml:22]: I will update the PR using ubuntu 20, It works

File: .github/workflows/publish-ecr.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Publish to ECR
+        env:
+          IMAGE_TAG: ${{steps.vars.outputs.tag}}
+          REP_NAME: trivy-db
+          ALIAS: aquasecurity
+        run: |
+          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/$ALIAS
+          docker build -t $REP_NAME:$IMAGE_TAG .
+          docker tag $REP_NAME:$IMAGE_TAG public.ecr.aws/$ALIAS/$REP_NAME:$IMAGE_TAG
+          docker push public.ecr.aws/$ALIAS/$REP_NAME:$IMAGE_TAG
```
[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/publish-ecr.yml:37]: yeah, definitely I will try again to simplify the code

[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/publish-ecr.yml]: @knqyf263 review this publish.yml is sync now with ecr and docker

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+---
+env:
+  - CGO_ENABLED=0
+  - GO111MODULE=on
+before:
+  hooks:
+    - go mod download
+builds:
+  - id: project
+    main: ./cmd/trivy-db
```
Comment (krol3) [inline .github/workflows/release.yml:10]: Adding release, the test is [here](https://github.com/krol3/trivy-db/releases/tag/v1.1.5)

Comment (knqyf263) [reply .github/workflows/release.yml:10]: It is unnecessary to release the trivy-db binary. We should release only the database file.

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          registry: public.ecr.aws
+          username: ${{ secrets.ECR_ACCESS_KEY_ID }}
+          password: ${{ secrets.ECR_SECRET_ACCESS_KEY }}
+      - name: Get the version
+        id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v1
+        with:
+          images: ${{ env.REP }}
+          tag-semver: |
+            {{version}}
```
Comment (knqyf263) [inline .github/workflows/publish.yml:46]: The DB version is like v1-2020122200. Does it work with `tag-semver`? I don't think v1-2020122200 is compatible with semver.

File: .github/workflows/cron.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply .github/workflows/cron.yml:1]: Yes, I know the linter complains about it, but I meant why the linter does that.

[the remark this answers is not in the document]
Comment (krol3) [reply .github/workflows/cron.yml:1]: not answer for that yet :)

File: .github/workflows/publish.yml (hunk 6 above)
Comment (krol3) [reply .github/workflows/publish.yml:51]: I use date to create the tag format

**Comment by knqyf263** [comment]
Thanks!
